teps/0086-changing-the-way-result-parameters-are-stored.md

```diff
@@ -117,6 +117,93 @@ Additionally, this will help projects that wrap/abstract Tekton where users unde
 
 ## Alternatives
 
+### Dedicated Storage API Service
+
+In this approach, a dedicated Result Storage API is used to abstract away large
+result storage and implementation.
+
+```
+┌───────────────────┐
+│                   │     ┌─────────────────────────┐
+│   Control Plane   │     │           Pod           │
+│                   │     │                         │
+│   ┌────────────┐  │     │ ┌──────────┐    ┌────┐  │
+│   │ Controller ├──┼─────┼─►entrypoint├────►step│  │
+│   └────────────┘  │     │ └─────┬────┘    └────┘  │
+│                   │     │       │                 │
+│                   │     │       │                 │
+│   ┌───────────┐   │     │       │                 │
+│   │Storage API◄───┼─────┼───────┘                 │
+│   └─────┬─────┘   │     │                         │
+│         │         │     └─────────────────────────┘
+└─────────┼─────────┘
+          │
+          │
+          │
+  ┌───────▼───────┐
+  │Storage Backend│
+  └───────────────┘
+```
+
+This approach relies on the fact that the entrypoint wraps the running user
+container. Today, Tekton utilizes this today to
+[order user steps by reading/writing post files](https://github.com/tektoncd/pipeline/blob/main/cmd/entrypoint/README.md),
+we could utilize this same process to perform other post-step actions on the
+Pod - i.e. we could use this to perform result uploading with the existing user
+Pod.
+
+The Storage API should be a known address to the entrypoint (either a static
```

Comment:
@wlynch since I'm new to `Tekton/Kubernetes` land, I had a simple question.
Is the `Storage API` a CRD? If not how does one go about implementing a custom API in the control plane?

Reply:
Great question! So in this case I don't think we'd go with a CRD, since that would eventually end up being bound by the same etcd limits that we face in Run CRDs today.

We'd probably just want an API running as a [k8s Service](https://kubernetes.io/docs/tutorials/kubernetes-basics/expose/expose-intro/) backed by a Deployment - for the reference implementation a ClusterIP service should be fine. HTTP/REST is probably easiest to implement with minimal dependencies, though other means (i.e. protobuf, etc.) could also be supported as long as both the Storage API and Entrypoint are compatible.

This would be similar to how the [Result API](https://github.com/tektoncd/results/blob/main/config/api.yaml) operates (separately - we should look into detangling these names at some point 😅 )